### PR TITLE
v0.2.0: Numbers and transactions utils

### PR DESCRIPTION
**Pull request by nop33**
The inspiration for this PR came when while working on the extension wallet I noticed a bug in the `formatAmountForDisplay` function when the input was exactly 1'000'000. I took the opportunity to move some functions from the desktop wallet to the SDK, so that they can be reused by the extension wallet, the mobile wallet, as well as by the explorer. Since these changes include a breaking change, I propose we bump to `0.2.0`.

## 💥 Breaking changes
- Rename `calAmountDelta` to `calcTxAmountDeltaForAddress` and improve it to take into account consolidation txs
  - _Note: I know I could have avoided the breaking changes by not renaming the function, but I really think this rename adds a lot of value and clarity as to what the function is actually doing, and it's something I wanted to do for a long time 😇 ._

## 🆕  New features
- Export `NUM_OF_ZEROS_IN_QUINTILLION`
- Export `produceZeros`
- Export `formatFiatAmountForDisplay`
- Export `convertSetToFiat`
- Create `isConsolidationTx` function
- Create `getDirection` function
- Create `removeConsolidationChangeAmount` function
- Add types `TransactionInfoType` and `TransactionDirection`
- Create simple `uniq` function to avoid installing `lodash` for it

## 🐛  Bug fixing
- Fix display of 1 million for `formatAmountForDisplay`
- Fix display of rounded up numbers for `formatAmountForDisplay`

### CODE REVIEW

**Review by mvaivre — APPROVED** [review]
LGTM! Appreciated changes.

File: test/numbers-test.ts [hunk 1]
```diff
@@ -153,6 +185,44 @@ it('should convert Set amount amount to Alph amount', () => {
...
+    expect(convertSetToFiat(BigInt('100000000000000000'), 2)).toEqual(0.2),
+    expect(convertSetToFiat(BigInt('10000000000000000'), 2)).toEqual(0.02),
+    expect(convertSetToFiat(BigInt('1000000000000000'), 2)).toEqual(0.002),
+    expect(convertSetToFiat(BigInt('100000000000000'), 2)).toEqual(0.0002),
+    expect(convertSetToFiat(BigInt('10000000000000'), 2)).toEqual(0.00002),
+    expect(convertSetToFiat(BigInt('1000000000000'), 2)).toEqual(0.000002),
+    expect(convertSetToFiat(BigInt('100000000000'), 2)).toEqual(0.0000002),
+    expect(convertSetToFiat(BigInt('10000000000'), 2)).toEqual(0.00000002),
+    expect(convertSetToFiat(BigInt('1000000000'), 2)).toEqual(0.000000002),
+    expect(convertSetToFiat(BigInt('1000000000'), 2)).toEqual(0.000000002),
```
Comment (mvaivre) [inline test/numbers-test.ts:199]: Duplicate of the line above

Comment (nop33) [reply test/numbers-test.ts]: Removed!

**Comment by nop33** [comment]
Got approval on Slack 😉

**Pull request merged**